scipy/special/_logit.h

```diff
@@ -3,12 +3,38 @@
 
 #include "numpy/npy_math.h"
 
-npy_float logitf(npy_float x);
-npy_double logit(npy_double x);
-npy_longdouble logitl(npy_longdouble x);
 
-npy_float expitf(npy_float x);
-npy_double expit(npy_double x);
-npy_longdouble expitl(npy_longdouble x);
+template <typename T>
+inline T _logit(T x) {
+    x /= 1 - x;
+    return npy_log(x);
```

Comment:
```std::log(x)```

Reply:
So we're not concerned with edge cases in the standard library implementations? 
It's full well possible I'm on a FUD side here, and am happy to follow an informed opinion, 
so let's have a more visible record: https://mail.python.org/pipermail/scipy-dev/2021-April/024738.html

Reply:
The main issue is that `std::log` is overloaded for different data types, while `npy_log` is `double`-only.  Note that the .src.c code calls `npy_log@c@`. So whatever is the outcome of the mailing list discussion, this will need to be changed either way.

Reply:
Ok, fair enough. Thanks. Am labeling this PR as needs-work until the mailing list discussion concludes (or will switch to `std::` versions if it's silent.)


Two comments

- the bug this PR introduces is undertested: https://github.com/scipy/scipy/blob/master/scipy/special/tests/test_logit.py looks like it would catch these things but the CI's green.
- this particular fix is going to fit in a single screen, so it'll be fine; but generally the way it all unfolds does make tempita templating more and more attractive.

Reply:
Unfortunately you can't test `long double` to higher than `double` precision because on some platforms (e.g. windows) `long double` doesn't offer any more precision than `double`.

Reply:
What would be a proper guard to xfail a test? Preferably not a platform tag, something like `sizeof(double) == sizeof(long double) `.

Reply:
Another problem is that on AIX I believe they use a non-standard format for extending precsion. If you need to know for sure that it's an 80-bit float, you could take this route:
https://github.com/scipy/scipy/blob/7bd68dff6cbea8c7189b4c76828a9fea9b633ee6/scipy/fft/_pocketfft/tests/test_real_transforms.py#L17-L24

Reply:
I remembered another detail from `scipy.fft`. Using GCC under windows subsystem for linux compiles with native 80 bit extended precision floats, but at runtime it still didn't improve precision for some complicated reason. It really was a mess to try to test for anything more than double precision.

Reply:
Ouch. What a mess. This pushes me still further down the "double precision should be enough for everybody" (tm) camp, I guess.